### PR TITLE
database connection url support

### PR DESCRIPTION
**Pull request by avelino**
refs:
https://github.com/prest/prest/issues/315
https://github.com/prest/prest/issues/258

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -141,6 +145,34 @@ func Parse(cfg *Prest) (err error) {
...
 	cfg.SSLRootCert = viper.GetString("ssl.rootcert")
+	if os.Getenv("DATABASE_URL") != "" {
+		// cloud factor support: https://devcenter.heroku.com/changelog-items/438
+		cfg.PGURL = os.Getenv("DATABASE_URL")
+	}
+	if cfg.PGURL != "" {
+		// Parser PG URL, get database connection via string URL
+		u, err := url.Parse(cfg.PGURL)
+		if err != nil {
+			return err
```
Comment (felipeweb) [inline config.go]: The function uses  named returning please do not mix the ways to returning

File: config.go [hunk 2]
```diff
@@ -141,6 +145,34 @@ func Parse(cfg *Prest) (err error) {
...
+			cfg.PGPort = pgPort
+		}
+		cfg.PGUser = u.User.Username()
+		pgPass, pgPassExist := u.User.Password()
+		if pgPassExist {
+			cfg.PGPass = pgPass
+		}
+		cfg.PGDatabase = strings.Replace(u.Path, "/", "", -1)
+		if len(u.Query()["sslmode"]) > 0 {
+			cfg.SSLMode = u.Query()["sslmode"][0]
```
Comment (felipeweb) [inline config.go]: can be used u.Query().Get(“sslmode”)

**Review by felipeweb — CHANGES_REQUESTED** [review]
Just a few comments

**Review by crgimenes — APPROVED** [review]
LGTM

**Review by felipeweb — APPROVED** [review]

**Pull request merged**